src/lib/nnlib.jl

```diff
@@ -1,5 +1,10 @@
 using NNlib
-import NNlib: softmax, ∇softmax, logsoftmax, ∇logsoftmax, conv, depthwiseconv, ∇conv_data, ∇depthwiseconv_data, maxpool, meanpool, σ
+import NNlib: softmax, ∇softmax, logsoftmax, ∇logsoftmax, conv, depthwiseconv, ∇conv_data, ∇depthwiseconv_data, maxpool, meanpool, σ, relu
+
+@adjoint function Base.Broadcast.broadcasted(::typeof(relu), xs::AbstractArray{T}) where T<:Number
+    ys = relu.(xs)
+    ys, Δ -> (nothing, broadcast((y, d) -> y > 0 ? d : zero(T), ys, Δ))
```

Comment:
I'd prefer this to be written as `ifelse.(...)`, rather than an explicit `broadcast` function call. Also, it may as well be `zero(y)` rather than `zero(T)`.

Reply:
Do you mean

```julia
@adjoint function Base.Broadcast.broadcasted(::typeof(relu), x::Numeric)
    y = relu.(x)
    y, Δ -> (nothing, ifelse.(y .> 0, Δ, zero(y)))
end
```
?
I know that the above code will cause extra memory allocation (`y .> 0` and `zero(y)`, although we can use `FillArray` for the latter). But sorry that I'm not very familiar with broadcast, I can't come up with a more efficient code using `ifelse.(...)`.

Reply:
If you use `zero.(y)`, both of those intermediates will actually get compiled away, so it's just as good.

Reply:
Ah, I see. Thanks!